src/main/java/surrealchat/command/DeleteCommand.java

```diff
@@ -0,0 +1,78 @@
+package surrealchat.command;
+
+import java.util.NoSuchElementException;
+
+import surrealchat.task.Task;
+import surrealchat.task.TaskManagement;
+
+/**
+ * Command object for deleting a Task object.
+ */
+public class DeleteCommand extends Command {
+    protected final String taskNumberString;
+
+    /**
+     * Creates new DeleteCommand object.
+     * @param taskNumberString Index number of Task to be deleted.
+     */
+    public DeleteCommand(String taskNumberString) {
+        super("delete");
+        this.taskNumberString = taskNumberString;
+    }
+
+    private int getInputNumber(String description) {
+        description = description.trim();
+
+        if (description.isEmpty()) {
+            throw new NoSuchElementException(
+                    "Did you forget to put a number for the command you just typed in? Not stonks!\n");
+        }
+
+        try {
+            return Integer.valueOf(description);
+        } catch (NumberFormatException e) {
+            throw new NumberFormatException(
+                    "Did you put something other than a number or did you put a number incorrectly? Not stonks!\n");
+        }
+    }
+
+    private boolean checkInvalidTaskNumber(int taskNumber, TaskManagement taskManagement) {
```

Comment:
instead of checkInvalidTaskNumber, perhaps you can rename is to isInvalidTaskNumber.